### DISCUSSION TITLE
RequestPermission does not pass when using SOG02.（SOG02　使用時に　requestPermissionが通らない。）

### DISCUSSION THREAD

**Discussion by HirokatsuIshikawa** [airsdk/Adobe-Runtime-Support#1713] [General]
There was a report from the user, but when using the following terminals, the file list suddenly became unable to open.

The cause seems to be the part using File requestPermission, and it seems that the event is not returned on the user's terminal even though it returns firmly on its own terminal.

If you don't have the file, you'll get another response, so that doesn't seem to be a problem.
Do you have any idea?

ユーザーからの報告があったのですが、以下の端末を使用した際に、突然ファイル一覧が開けなくなったとの事です。

原因はFileのrequestPermissionを使用している箇所のようで、自身の端末ではしっかり返ってくるのに、ユーザーの端末ではイベントが返ってこないそうです。
ファイルが無い場合は、別の反応が返ってくるため、それが問題とは思えません。
何か心当たりはございませんでしょうか？

            _file = dir.resolvePath(CommonSystem.FOLDER_NAME + "//Scenario");
            _file.addEventListener(PermissionEvent.PERMISSION_STATUS, checkPermissions);
            
            try
            {
                _file.requestPermission();
            }
            catch (e:PermissionError)
            {
                // another request is in progress
                trace("REQUEST ERROR!!! : " + e.toString());
            }

*SOG2
https://www.au.com/mobile/product/smartphone/sog02/

* Target application / free version
※対象のアプリ・無料版

https://play.google.com/store/apps/details?id=air.air.NeoSRCMobile&hl=ja

**Comment by lwmirkk** [comment]
Are you using what SDK VERSION?
33.1.1.779?

**Comment by HirokatsuIshikawa** [comment]
Yes.

At least I updated it with the latest version during the last week.

はい。

少なくとも、先週中に最新版で更新しました。

**Comment by lwmirkk** [comment]
I think SOG02 uses Android 11, so probably it is related to this problem:
https://github.com/airsdk/Adobe-Runtime-Support/issues/1714

**Comment by HirokatsuIshikawa** [comment]
I'm late, but thank you very much for your reply.

In that case, if you are using Android 11 or later, is there any way to read the area that can be written from the outside?

Because of the specifications of this app, this method is absolutely necessary ...

遅れましたが、返信どうもありがとうございます。

そうなると、Android11以降を使用している場合に、外から書き込みできる領域を読み込む方法は外にはないものでしょうか？

このアプリの仕様上、どうしてもこの方法が必要になってしまうので……。

**Reply by lwmirkk** [reply]
Sorry, I don't know how to proceed with this, maybe @ajwfrost or @marchbold know something in how to save to root storage outside applicationDirectory. =)

**Reply by HirokatsuIshikawa** [reply]
情報ありがとうございます。
あれから追加で色々調べたところ、パーミッションにMANAGE_EXTERNAL_STORAGEを追加すればうまく行く感じのようでした。
ただし、GooglePlayで申請する際に申請が必要になります。

ご丁寧にどうもありがとうございました。

Thank you for the information.
After doing some additional research, it seemed like adding MANAGE_EXTERNAL_STORAGE to the permissions would work.
However, you will need to apply when applying on Google Play.

Thank you very much for your kindness.

**Comment by HirokatsuIshikawa** [comment]
追加で失礼します。

申請通りませんでした、ちょっとこの方法だとまずいみたいですね……

Excuse me for the addition.

I didn't meet the application, it seems that this method is not good ...

**Reply by lwmirkk** [reply]
It did not work?
Maybe @ajwfrost can help you how you could access and save files to root directory in Android 10/11.

**Reply by HirokatsuIshikawa** [reply]
No, I just didn't pass the Google Play review.

If I look for it a little more, I feel that it is quite likely, so I will proceed with the development while referring to the information I received.

thank you very much.

いえ、GooglePlayの審査に通らなかったたけです。

もう少し探せば何とかなりそうな気がしてきたので、いただいた情報を参考にしつつ開発を進めさせていただきます。

どうもありがとうございました。
